tests/test_class_basics.rs

```diff
@@ -456,9 +456,8 @@ fn dunder_dict_support() {
     });
 }
 
-// Accessing inst.__dict__ only supported in limited API from Python 3.10
 #[test]
-#[cfg_attr(all(Py_LIMITED_API, not(Py_3_10)), ignore)]
+#[cfg(any(Py_3_9, not(Py_LIMITED_API)))]
```

Comment:
I always get confused a bit when looking at these cfg combinations 😅, but I feel like we don't need any gates for these tests anymore. At least from the implementation it looks like we always push a `PyGetSetDef` now.

Reply:
I believe the gate is still needed due to `#[pyclass(dict)]` not being supported on 3.7 / 3.8. But I think you're right, we are incorrectly pushing `PyGetSetDef` on those versions, I'll correct that.

Reply:
Ah, I see. In that case the bound (and the newsfragment) makes sense to me.